File: JS101_programming_foundations/small_problems/05_easy_5/ex_08.js
// Write a function that takes one argument, a positive integer, and returns a list of the digits in the number.

//map
function digitList(digit) {
  return String(digit).split('').map ((num) => {
    return Number(num);
  });
}

// function digitList(digit) {
//   let numbers = [];
//   while (digit > 0) {
//     let remainder = digit % 10;
//     numbers.unshift(remainder);
//     digit = Math.floor(digit / 10);
//   }
//   console.log(numbers);
// }

digitList(12345);       // [1, 2, 3, 4, 5]
digitList(7);           // [7]
digitList(375290);      // [3, 7, 5, 2, 9, 0]
digitList(444);         // [4, 4, 4]

File: RB101_programming_foundations/exercises/medium_2/ex_08.rb
# A featured number (something unique to this exercise) is an odd number that is a multiple of 7, and whose digits occur exactly once each. So, for example, 49 is a featured number, but 98 is not (it is not odd), 97 is not (it is not a multiple of 7), and 133 is not (the digit 3 appears twice).

# Write a method that takes a single integer as an argument, and returns the next featured number that is greater than the argument. Issue an error message if there is no next featured number.

def featured(num)
  num = is_odd_divisible_7(num)
  loop do
    return num if no_repeated_digit?(num)
    num += 14
    break if num >= 9_999_999_999
  end
  puts "There is no possible number that fulfills those requirements"
end

def is_odd_divisible_7(num)
  loop do
    num += 1
    break if num.odd? && num % 7 == 0
  end
  num
end

def no_repeated_digit?(num)
  num_array = num.to_s.split('')
  return true if num_array.uniq == num_array
  false
end

featured(12) == 21
featured(20) == 21
featured(21) == 35
featured(997) == 1029
featured(1029) == 1043
featured(999_999) == 1_023_547
featured(999_999_987) == 1_023_456_987

featured(9_999_999_999) # -> There is no possible number that fulfills those requirements
